src/main/java/org/mskcc/cbio/portal/dao/DaoSampleList.java

```diff
@@ -54,15 +54,22 @@ public int addSampleList(SampleList sampleList) throws DaoException {
             con = JdbcUtil.getDbConnection(DaoSampleList.class);
 
             pstmt = con.prepareStatement("INSERT INTO sample_list (`STABLE_ID`, `CANCER_STUDY_ID`, `NAME`, `CATEGORY`," +
-                    "`DESCRIPTION`)" + " VALUES (?,?,?,?,?)");
+                    "`DESCRIPTION`)" + " VALUES (?,?,?,?,?)", Statement.RETURN_GENERATED_KEYS);
             pstmt.setString(1, sampleList.getStableId());
             pstmt.setInt(2, sampleList.getCancerStudyId());
             pstmt.setString(3, sampleList.getName());
             pstmt.setString(4, sampleList.getSampleListCategory().getCategory());
             pstmt.setString(5, sampleList.getDescription());
             rows = pstmt.executeUpdate();
-   			int listListRow = addSampleListList(sampleList, con);
-   			rows = (listListRow != -1) ? (rows + listListRow) : rows;
+            try (ResultSet generatedKey = pstmt.getGeneratedKeys()) {
```

Comment:
I don't think this nested try is necessary. pstmt.getGeneratedKeys() throws SQLException so the higher level catch will catch this exception.

Reply:
this is try-with-resources to close the result set at the end of the block. It does not catch anything.